File: web/src/lib/date-format.ts
/** Human-readable rendering of a date property value. Accepts a bare
 *  `YYYY-MM-DD` (optionally ` HH:mm`) or a `[[YYYY-MM-DD]]`-wrapped value.
 *  Unrecognized input is returned trimmed-but-unchanged. */
export function formatDateMonthDay(v: string): string {
  // Accept `[[YYYY-MM-DD]]` (optionally followed by ` HH:mm`) and bare ISO.
  const m = v.trim().match(/^\[\[(\d{4})-(\d{2})-(\d{2})\]\](?:\s+(\d{2}):(\d{2}))?$/) ||
            v.trim().match(/^(\d{4})-(\d{2})-(\d{2})(?:\s+(\d{2}):(\d{2}))?$/);
  if (!m) return v.trim();
  const [, y, mo, d, hh, mm] = m;
  const date = new Date(Number(y), Number(mo) - 1, Number(d));
  const month = date.toLocaleString("en-US", { month: "short" });
  const day = Number(d);
  const thisYear = new Date().getFullYear();
  const datePart = Number(y) === thisYear ? `${month} ${day}` : `${month} ${day}, ${y}`;
  if (!hh) return datePart;
  let h = Number(hh);
  const ampm = h >= 12 ? "p" : "a";
  h = h % 12 || 12;
  const minStr = mm === "00" ? "" : `:${mm}`;
  return `${datePart} ${h}${minStr}${ampm}`;
}

File: app/Tesela-iOS/Sources/Data/DateFormat.swift
import Foundation

enum DateFormat {
    /// Human-readable rendering of a date property value. Accepts bare
    /// `YYYY-MM-DD` (optionally ` HH:mm`) or legacy `[[YYYY-MM-DD]]`.
    /// Unrecognized input returned trimmed-unchanged.
    /// Mirrors `web/src/lib/date-format.ts::formatDateMonthDay`.
    static func humanMonthDay(_ v: String) -> String {
        let trimmed = v.trimmingCharacters(in: .whitespaces)
        let patterns = [
            #"^\[\[(\d{4})-(\d{2})-(\d{2})\]\](?:\s+(\d{2}):(\d{2}))?$"#,
            #"^(\d{4})-(\d{2})-(\d{2})(?:\s+(\d{2}):(\d{2}))?$"#,
        ]
        for p in patterns {
            let re = try! NSRegularExpression(pattern: p)
            guard let m = re.firstMatch(in: trimmed, range: NSRange(trimmed.startIndex..., in: trimmed)) else { continue }
            guard let yr = Range(m.range(at: 1), in: trimmed),
                  let mr = Range(m.range(at: 2), in: trimmed),
                  let dr = Range(m.range(at: 3), in: trimmed) else { continue }
            let y = Int(trimmed[yr])!, mo = Int(trimmed[mr])!, d = Int(trimmed[dr])!
            var dc = DateComponents(); dc.year = y; dc.month = mo; dc.day = d
            guard let date = Calendar(identifier: .gregorian).date(from: dc) else { return trimmed }
            let monthFmt = DateFormatter()
            monthFmt.locale = Locale(identifier: "en_US_POSIX")
            monthFmt.dateFormat = "MMM"
            let monthLabel = monthFmt.string(from: date)
            let thisYear = Calendar.current.component(.year, from: Date())
            let dateStr = y == thisYear ? "\(monthLabel) \(d)" : "\(monthLabel) \(d), \(y)"

            // Optional time — only present when both HH and MM matched (groups 4 and 5).
            if m.numberOfRanges >= 6,
               m.range(at: 4).location != NSNotFound,
               let hr = Range(m.range(at: 4), in: trimmed),
               let mr2 = Range(m.range(at: 5), in: trimmed) {
                let hh = Int(trimmed[hr])!, mm = Int(trimmed[mr2])!
                let ampm = hh >= 12 ? "p" : "a"
                var h = hh % 12
                if h == 0 { h = 12 }
                let minStr = mm == 0 ? "" : String(format: ":%02d", mm)
                return "\(dateStr) \(h)\(minStr)\(ampm)"
            }
            return dateStr
        }
        return trimmed
    }
}
